### PR TITLE
Apply breaking changes of new libp2p versions

### PR DESCRIPTION
**Pull request by tomaka**
The master branch of libp2p contains breaking changes. This PR updates Substrate for these changes.

This PR is only to be merged once a new version is published, which will likely happen in the next two weeks.

### CODE REVIEW

**Comment by gavofyork** [comment]
I'm not sure what kind of breakage this refers to. If it's wire-line breakage, then it should include a full migration plan and implementation. If it's just APIs, then it will need a corresponding PR in polkadot codebase.

**Comment by tomaka** [comment]
It's only the API of libp2p which has changed. Nothing has changed on the wire.
Polkadot doesn't directly depend on libp2p, and Substrate's API is untouched, so there's no Polkadot PR to do.

**Comment by tomaka** [comment]
We published libp2p 0.13.0, so this is now ready for review/merge.

File: core/network/src/debug_info.rs [hunk 1]
```diff
@@ -287,16 +287,15 @@ where TSubstream: AsyncRead + AsyncWrite {
...
-						IdentifyEvent::Identified { peer_id, info, .. } => {
+						IdentifyEvent::Received { peer_id, info, .. } => {
 							self.handle_identify_report(&peer_id, &info);
 							let event = DebugInfoEvent::Identified { peer_id, info };
 							return Async::Ready(NetworkBehaviourAction::GenerateEvent(event));
 						}
-						IdentifyEvent::Error { .. } => {}
-						IdentifyEvent::SendBack { result: Err(ref err), ref peer_id } =>
+						IdentifyEvent::Error { peer_id, error } =>
 							debug!(target: "sub-libp2p", "Error when sending back identify info \
```
Comment (romanb) [inline core/network/src/debug_info.rs]:
```suggestion
							debug!(target: "sub-libp2p", "Identification with peer {:?} failed => {}", peer_id, error)
```
`IdentifyEvent::Error` is no longer specific to errors occurring during sending of replies.

**Pull request merged**